src/Components/Updater/Provider.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Laravel Zero.
+ *
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace LaravelZero\Framework\Components\Updater;
+
+use function class_exists;
+use Illuminate\Console\Application as Artisan;
+use LaravelZero\Framework\Components\AbstractComponentProvider;
+
+/**
+ * @internal
+ */
+final class Provider extends AbstractComponentProvider
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isAvailable(): bool
+    {
+        return class_exists(\Humbug\SelfUpdate\Updater::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function register(): void
+    {
+        $config = $this->app['config'];
+
+        if ($config->get('app.production', false)) {
```

Comment:
You need to check if your are in `build` mode. @ijpatricio have a class for this. it's @ijpatricio merged?

Reply:
Ok, I'll check if it's merged or not, cause this code is based entirely on the `stable` branch, not the `develop` one

Reply:
I think its not merged yet https://github.com/laravel-zero/framework/pull/312

Reply:
@linuxjuggler It's merged now. You can rebase and get the class. It's Providers\Build\Build.php

Reply:
let me check and see what can I enhance.

Reply:
sorry to bother you @nunomaduro , I just revisit the code and now we check only for two things:

1. Is it a production environment
2. are we executing the command from within a phar file.